### ISSUE TITLE
Better documentation for loss functions

### ISSUE THREAD

**Issue by lcrmorin** [microsoft/LightGBM#4790]
## Summary

Loss function documentation currently send to wikipedia & kaggle. It's not clear how parameters (alpha for huber, quantile loss and c for fair loss) play. It's not clear what range are acceptable for these parameters.

## Motivation

Better documentation for loss functions would help their usage and adoption. 

## Description

Some documentation of the loss, including: formula, theoretical aspects and parameters would be nice. 

Ps: I am willing to participate in this documentation. But I am not familiar with the process of updating said documentation.

**Comment by jameslamb** [comment]
Thanks very much for using LightGBM and for your interest in this.

Are you looking for LightGBM to maintain documentation on each objective function with a similar level of detail to `scikit-learn`'s?

For example, https://scikit-learn.org/stable/modules/model_evaluation.html#mean-tweedie-deviance 

------------------------

Separate from that general question, we'd be very grateful for contributions that update the documentation for other parameters that are specific to individual objectives.

I agree with your assessment...documentation for `fair_c`, for example, doesn't currently explain these key things:

* what is the scale of this parameter?
* what are the implications of increasing or decreasing it?

I'd be happy to help you through the contributions, if you'd like! The process involves updating comments in a header file, which we use to code-generate the ReStructuredText used in the documentation site.

For example, for `fair_c` you would add to this block:

https://github.com/microsoft/LightGBM/blob/6cbb3586b92112fc1f27a179b201ac924e5600cc/include/LightGBM/config.h#L861-L864

then run `python helpers/parameter_generator.py`

If you choose to help with this, we'd prefer one pull request per parameter, to keep the PRs small and easy to review.

**Comment by StrikerRUS** [comment]
Closed in favor of being in #2302. We decided to keep all feature requests in one place.

Welcome to contribute this feature! Please re-open this issue (or post a comment if you are not a topic starter) if you are actively working on implementing this feature.

**Comment by jameslamb** [comment]
Sorry, this was locked accidentally. Just unlocked it. We'd still love help with this feature!
